gorb/gorb.go

```diff
@@ -0,0 +1,323 @@
+/*
+Package gorb provide the ability to register and deregister the backend
+*/
+package gorb
+
+import (
+	"errors"
+
+	"time"
+
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"io"
+
+	"path"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/hashicorp/go-multierror"
+	"github.com/sky-uk/feed/controller"
+)
+
+// HealthcheckArgs defines health check URI
+type HealthcheckArgs struct {
+	Method string `json:"method"`
+	Path   string `json:"path"`
+	Expect string `json:"expect"`
+}
+
+// Pulse defines backend health check
+type Pulse struct {
+	TypeHealthcheck string          `json:"type"`
+	Args            HealthcheckArgs `json:"args"`
+	Interval        string          `json:"interval"`
+}
+
+// BackendConfig defines the backend configuration
+type BackendConfig struct {
+	Host   string `json:"host"`
+	Port   int    `json:"port"`
+	Method string `json:"method"`
+	Weight int    `json:"weight"`
+	Pulse  Pulse  `json:"pulse"`
+}
+
+// Backend defines the backend configuration + service name
+type backend struct {
+	ServiceName   string
+	BackendConfig BackendConfig
+}
+
+type loopbackAction string
+
+const (
+	addLoopback    loopbackAction = "add"
+	deleteLoopback loopbackAction = "del"
+)
+
+// New creates a gorb handler
+func New(serverBaseURL string, instanceIP string, drainDelay time.Duration, servicesDefinition string, backendWeight int, backendMethod string, vipLoadbalancer string, manageLoopback bool, backendHealthcheckInterval string, interfaceProcFsPath string) (controller.Updater, error) {
+	if serverBaseURL == "" {
+		return nil, errors.New("unable to create Gorb updater: missing server ip address")
+	}
+	initMetrics()
+	log.Infof("Gorb server url: %s, drainDelay: %v, instance ip adddress: %s, vipLoadbalancer: %s ", serverBaseURL, drainDelay, instanceIP, vipLoadbalancer)
+
+	backendDefinitions := []backend{}
+	servicesDefinitionArr := strings.Split(servicesDefinition, ",")
```

Comment:
I'd still rather this and `servicesArr` be passed in as slices so we can type check rather than relying on string parsing.